From 77450953af8a3d3080f716c08e0fb9295fb9098b Mon Sep 17 00:00:00 2001
From: Mika Laitio <mika.laitio@amd.com>
Date: Tue, 10 Jun 2025 17:37:18 -0700
Subject: [PATCH] therock install and launch scripts for comfyui

1. thr_install.sh is invoked by the rockbuilders
  comfyui.cfg to install the comfyui to directory
  in $ROCM_HOME/apps/comfyui
  This script will take two parameters:
  - location containing the git clone of comfyui sources
  - ROCM_HOME directory

2. thr_comfyui.sh is the launcher script that will start
  the comfyui from it's install location
  ($ROCM_HOME/apps/comfyui)

Signed-off-by: Mika Laitio <mika.laitio@amd.com>
---
 thr_comfyui.sh | 11 +++++++++++
 thr_install.sh | 20 ++++++++++++++++++++
 2 files changed, 31 insertions(+)
 create mode 100644 thr_comfyui.sh
 create mode 100755 thr_install.sh

diff --git a/thr_comfyui.sh b/thr_comfyui.sh
new file mode 100644
index 00000000..e5a503b8
--- /dev/null
+++ b/thr_comfyui.sh
@@ -0,0 +1,11 @@
+if [ -z "$ROCM_HOME" ]; then
+    echo "Error, ROCM_HOME not specified."
+    exit 1
+fi
+if test -d "${ROCM_HOME}/apps/comfyui"; then
+    cd ${ROCM_HOME}/apps/comfyui
+    export LD_LIBRARY_PATH=${ROCM_HOME}/lib:${ROCM_HOME}/lib/llvm/lib:${LD_LIBRARY_PATH}
+    python main.py
+else
+    echo "Could not find comfyui install from ${ROCM_HOME}/apps/comfyui"
+fi
diff --git a/thr_install.sh b/thr_install.sh
new file mode 100755
index 00000000..df15390f
--- /dev/null
+++ b/thr_install.sh
@@ -0,0 +1,20 @@
+if [ -z "$1" ]; then
+    echo "Error, application source directory not specified."
+    exit 1
+else
+    APP_SRC_DIR=${1}
+fi
+if [ -z "$2" ]; then
+    echo "Error, no rocm_root_directory parameter specified."
+    exit 1
+else
+    INSTALL_DIR_PREFIX_ROCM=${2}
+    echo "rocm_root_directory parameter: ${install_dir_prefix_rocm}"
+fi
+INSTALL_DIR_PREFIX_APP=${INSTALL_DIR_PREFIX_ROCM}/apps/comfyui
+
+mkdir -p ${INSTALL_DIR_PREFIX_APP}
+cd ${APP_SRC_DIR}
+git archive HEAD | tar -x -C ${INSTALL_DIR_PREFIX_APP}
+cp ${APP_SRC_DIR}/thr_comfyui.sh ${INSTALL_DIR_PREFIX_ROCM}/bin
+chmod u+x ${INSTALL_DIR_PREFIX_ROCM}/bin/thr_comfyui.sh
-- 
2.43.0

